### PR TITLE
[SPARK-34355][SQL] Add log and time cost for commit job

### PR DESCRIPTION
**Pull request by ulysses-you**
### What changes were proposed in this pull request?

Add some info log around commit log.

### Why are the changes needed?

Th commit job is a heavy option and we have seen many times Spark block at this code place due to the slow rpc with namenode or other.

It's better to record the time that commit job cost.

### Does this PR introduce _any_ user-facing change?

Yes, more info log.

### How was this patch tested?

Not need.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala
Comment (LuciferYang) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala:224]: Maybe should assign `System.nanoTime()` to a `val commitJobEndTime` first, otherwise the `Cost time` may not be accurate.

Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala:224]: Cost time -> Elapsed time?

Comment (ulysses-you) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala]: yea, that's more accurate.

Comment (ulysses-you) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala]: Updated.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala
Comment (yaooqinn) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala:224]: How about use ` val (_, duration) = Utils.timeTakenMs { committer.commitJob(job, commitMsgs) }`

Comment (dongjoon-hyun) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala:224]: +1 for @yaooqinn 's suggestion to use `Utils.timeTakenMs`.

Comment (ulysses-you) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala]: Followed this.

**Review by yaooqinn — APPROVED** [review]

**Review by HeartSaVioR — APPROVED** [review]
+1

**Comment by HeartSaVioR** [comment]
Actually I have been thinking about this - while I think this helps to track down the elapsed time on committing job, there's still another problem end users confuse the huge delay of commit job with being stuck unless they took the stack trace. Unfortunately Hadoop layer leaves log messages with DEBUG level, hence no log message may be written during commit phase. (I've added latency information on DEBUG log message in Hadoop side via MAPREDUCE-7317, but will be available in 3.3.1+ and the log level is still DEBUG.)

Probably we'd need to print out some informative log message during commit phase periodically - not sure where is the right place to fix, Hadoop vs Spark.

**Comment by ulysses-you** [comment]
@HeartSaVioR thanks for the sharing. Yea, agree. It's better to provide a progress-like stuff during commit job, but seems it's hard to do this at Spark side.

**Comment by HeartSaVioR** [comment]
OK. I'll leave this till early next week and merge if there's no further comment.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. I'll leave this to @HeartSaVioR since he has been waiting for this according to this comment.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala
Comment (AngersZhuuuu) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala:223]: +1

**Review by AngersZhuuuu — APPROVED** [review]

**Comment by HeartSaVioR** [comment]
retest this, please

**Comment by HeartSaVioR** [comment]
Will merge once either Jenkins or GA passes.

**Comment by HeartSaVioR** [comment]
Thanks! Merging to master.

**Comment by ulysses-you** [comment]
thanks all !

**Pull request closed without merging**
